eppy/function_helpers.py

```diff
@@ -85,36 +104,161 @@ def subsurfaces(ddtt):
     return ddtt.getreferingobjs(**kwargs)
 
 
+@register_epbunch_function("rvalue", keys=None)
+@property
 def rvalue(ddtt):
     return thermal_properties.rvalue(ddtt)
 
 
+@register_epbunch_function("ufactor", keys=None)
+@property
 def ufactor(ddtt):
     return thermal_properties.ufactor(ddtt)
 
 
+@register_epbunch_function("ufactor_ip", keys=None)
+@property
 def ufactor_ip(ddtt):
     return thermal_properties.ufactor_ip(ddtt)
 
 
+@register_epbunch_function("rvalue_ip", keys=None)
+@property
 def rvalue_ip(ddtt):
     return thermal_properties.rvalue_ip(ddtt)
 
 
+@register_epbunch_function("heatcapacity", keys=None)
+@property
 def heatcapacity(ddtt):
     return thermal_properties.heatcapacity(ddtt)
 
 
+@register_epbunch_function("f_fanpower_bhp", keys=None)
+@property
 def fanpower_bhp(ddtt):
     """return fanpower in bhp"""
     return eppy.fanpower.fanpower_bhp(ddtt)
 
 
+@register_epbunch_function("f_fanpower_watts", keys=None)
+@property
 def fanpower_watts(ddtt):
     """return fanpower in watts"""
     return eppy.fanpower.fanpower_watts(ddtt)
 
 
+@register_epbunch_function("f_fan_maxcfm", keys=None)
+@property
 def fan_maxcfm(ddtt):
     """return the Maximum_Flow_Rate in cfm"""
     return eppy.fanpower.fan_maxcfm(ddtt)
+
+
+def addfunctions(abunch):
+    """add functions to epbunch"""
+
+    key = abunch.obj[0].upper()
+
+    # -----------------
+    # TODO : alternate strategy to avoid listing the objkeys in snames
+    # check if epbunch has field "Zone_Name" or "Building_Surface_Name"
+    # and is in group u'Thermal Zones and Surfaces'
+    # then it is likely to be a surface.
+    # of course we need to recode for surfaces that do not have coordinates :-(
+    # or we can filter those out since they do not have
+    # the field "Number_of_Vertices"
```

Comment:
Filtering is fine, I think.

Reply:
This is for @santoshphilip I think. Not my code.